streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java

```diff
@@ -1247,4 +1270,33 @@ public void waitForNextStableAssignment(final long maxWaitMs) throws Interrupted
             );
         }
     }
+
+    public static class TrackingStateRestoreListener implements StateRestoreListener {
```

Comment:
This isn't threadsafe, but it looks like we're using it from multiple threads during the tests.

Reply:
Whoops